### ISSUE TITLE
Compiled Warlock renderer hangs on DOSBox (but runs on VMWare)

### ISSUE THREAD

**Issue by 64kramsystem** [joncampbell123/dosbox-x#3843]
### Describe the bug

Warlock is an education project from the book "Tricks of the game programming gurus" (not the demo!).

When compiled from sources, the resulting executable hangs on DOSBox, while it runs fine (super fast :smile:) on VMWare.

### Steps to reproduce the behaviour

Just run:

```sh
git clone git@github.com:64kramsystem/warlock-64k
cd warlock-64k
git checkout 1256d1d
wget https://github.com/joncampbell123/dosbox-x/files/10056653/WARLOCK.EXE.zip -O /tmp/warlock.exe.zip
unzip /tmp/warlock.exe.zip
perl -i -pe 's/COMPILE.BAT/WARLOCK.EXE/' dosbox-x.conf
dosbox-x
```

Running these commands will run the executable, which will hang. If you copy the project to a floppy, and run it on a VMWare machine with MS-DOS (I've tried with 5.0) installed, it will run fine.

The repository contains the dosbox configuration I've used.

In theory, you can also recompile the project (COMPILE.BAT automates everything), although it requires the toolchain (MASM + M$ C 7.00) to be installed.

### Expected behavior

The program should run (not hang).

### What operating system(s) this bug have occurred on?

Ubuntu 22.04

### What version(s) of DOSBox-X have this bug?

0.84.3, built from the sources.

### Used configuration

```ini
[sdl]
maximize = true

[dosbox]
fastbioslogo = true
startbanner = false
quit warning = false

[render]
aspect = true

[cpu]
cycles = 200000

[autoexec]
MOUNT C . -freesize 128
C:

C:\SUPPORT\132x50

WARLOCK.EXE
```
```


### Output log

```text
When I execute `WARLOCK.EXE`, the log reports:


LOG:  747876978 ERROR INT10:Function EF00 not supported
```
```


### Additional information

_No response_

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by 64kramsystem** [comment]
Note: I'm not sure if Github will keep around the file indefinitely (since the link to it is in a Markdown code block). If it happens to garbage collect it, let me know, and I'll reupload it and leave an explicit link in the Issue description.

**Comment by grapeli** [comment]
If a program doesn't work with a modified dosbox-x configuration, then it's worth checking with the **default** one first. If it works, check which modification affects the launch of the program.

`rem C:\SUPPORT\132x50`

**Comment by grapeli** [comment]
The 132x50 program is included with dosbox-x. It is located in z:\textutil.

You have two versions of `warlock.exe`.
The [one here](https://github.com/64kramsystem/warlock-64k/blob/main/support/WARLOCK.EXE) runs fine with 132x50.
This [one](https://github.com/joncampbell123/dosbox-x/files/10056653/WARLOCK.EXE.zip) isn't.

**Comment by 64kramsystem** [comment]
> The 132x50 program is included with dosbox-x. It is located in z:\textutil.
> 
> You have two versions of `warlock.exe`. The [one here](https://github.com/64kramsystem/warlock-64k/blob/main/support/WARLOCK.EXE) runs fine with 132x50. This [one](https://github.com/joncampbell123/dosbox-x/files/10056653/WARLOCK.EXE.zip) isn't.

Right. Not executing the 132x50 program makes the executable run fine (the one under `support/` can be ignored), at least on the machine I'm running now, so this is not a DOSBox issue - it has to do with 132x50.

I'm therefore closing the issue. Thanks!

**Comment by 64kramsystem** [comment]
> The 132x50 program is included with dosbox-x. It is located in z:\textutil.

Thanks for that, I wasn't aware! :blush:
